nautobot/utilities/forms/fields.py

```diff
@@ -666,3 +666,41 @@ def to_python(self, value):
         except ValueError as error:
             raise ValidationError(error)
         return super().to_python(value)
+
+
+class NaturalKeyMultipleChoiceField(django_filters.fields.ModelMultipleChoiceField):
+    """
+    Field to support matching an object's `pk` or `slug` field depending on the value
+    supplied. Faises ValidationError if neither field matches. Matches `pk` on the
+    `iexact` lookup expression or `slug` on `icontains`.
+    """
+
+    def _check_values(self, values):
+        """
+        This method overloads the grandparent method in `django.forms.models.ModelMultipleChoiceField`,
+        re-using some of that method's existing logic and adding support for coupling this field with
+        multiple model fields.
+        """
+        # deduplicate given values to avoid creating many querysets or
+        # requiring the database backend deduplicate efficiently.
+        try:
+            values = frozenset(values)
+        except TypeError:
+            # list of lists isn't hashable, for example
+            raise ValidationError(
+                self.error_messages["invalid_list"],
+                code="invalid_list",
+            )
+        for item in values:
+            qs = self.queryset.filter(Q(pk__iexact=str(item)) | Q(slug__icontains=str(item)))
+            if not qs.exists():
+                raise ValidationError(
+                    self.error_messages["invalid_choice"],
+                    code="invalid_choice",
+                    params={"value": item},
+                )
```

Comment:
Isn't this potentially really inefficient in terms of extra database queries? If I pass a list of 20 values, it's got to do 20 database queries just to decide if the values are valid?

Reply:
This is how it's done in the [grandparent django field](https://github.com/django/django/blob/08e6073f878264a0c091da0d3db456820252ef6c/django/forms/models.py#L1346). Maybe it's done this way so the invalid value can be shown in the error message?